drivers/usbhost/usbhost_bthci.c

```diff
@@ -0,0 +1,1536 @@
+ *
+ * Returned Value:
+ *   Zero is returned on success; a negated errno value is returned on any
+ *   failure.
+ *
+ ****************************************************************************/
+
+static int usbhost_bthci_open(FAR struct bt_driver_s *dev)
+{
+  return OK;
+}
+
+/****************************************************************************
+ * Name: usbhost_bthci_close
+ *
+ * Description:
+ *   Close the bluetooth hardware.
+ *
+ * Input Parameters:
+ *   dev   - An instance of the BT low-level driver's interface structure.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void usbhost_bthci_close(FAR struct bt_driver_s *dev)
+{
+}
+
+/****************************************************************************
+ * Name: usbhost_bthci_ioctl
+ *
+ * Description:
+ *   Handle IOCTL commands directed to this device.
+ *
+ * Input Parameters:
+ *   dev - An instance of the BT low-level driver's interface structure.
+ *   cmd - The IOCTL command
+ *   arg - The argument for the IOCTL command
+ *
+ * Returned Value:
+ *   Zero is returned on success; a negated errno value is returned on any
+ *   failure.
+ *
+ ****************************************************************************/
+
+static int usbhost_bthci_ioctl(FAR struct bt_driver_s *dev,
+                 int cmd, unsigned long arg)
+{
+  return -ENOTTY;
+}
+
+/****************************************************************************
+ * Name: usbhost_devinit
+ *
+ * Description:
+ *   The USB device has been successfully connected.  This completes the
+ *   initialization operations.  It is first called after the
+ *   configuration descriptor has been received.
+ *
+ *   This function is called from the connect() method.  This function always
+ *   executes on the thread of the caller of connect().
+ *
+ * Input Parameters:
+ *   priv - A reference to the class instance.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static inline int usbhost_devinit(FAR struct usbhost_state_s *priv)
+{
+  int ret = OK;
+
+  if (priv->bulkin && work_available(&priv->acwork))
+    {
+      /* Begin monitoring of acl packets */
+
+      work_queue(LPWORK, &priv->acwork, usbhost_acl_work, priv, 0);
+    }
+
+  if (priv->intin && work_available(&priv->evwork))
+    {
+      /* Begin monitoring of message available events */
+
+      work_queue(LPWORK, &priv->evwork, usbhost_event_work, priv, 0);
+    }
+
+  if (ret >= 0 && priv->devchar == 'a')
+    {
+      /* Register the driver with the network stack. */
+
+      ret = bt_netdev_register(&priv->btdev);
+      if (ret < 0)
+        {
+          uerr("ERROR: bt_netdev_register failed: %d\n", ret);
+        }
+    }
+
+  /* Check if we successfully initialized. */
+
+  if (ret >= 0)
+    {
+      ret = nxsem_wait_uninterruptible(&priv->exclsem);
```

Comment:
why need wait sem here

Reply:
Thanks. I removed the code I was protecting. I'll remove this.